kivy_garden/mapview/utils.py

```diff
@@ -43,8 +44,191 @@ def get_zoom_for_radius(radius_km, lat=None, tile_size=256.0):
     # Check how many tiles that are currently in view
     nr_tiles_shown = min(Window.size) / dp(tile_size)
 
-    # Keep zooming in until we find a zoom level where the circle can fit inside the screen
+    # Keep zooming in until we find a zoom level where the circle
+    # can fit inside the screen
     zoom = 1
-    while earth_circumference / (2 << (zoom - 1)) * nr_tiles_shown > 2 * radius:
-        zoom += 1
+    with suppress(OverflowError):
+        while (
+            earth_circumference
+            / (2 << (zoom - 1)) * nr_tiles_shown
+            > 2 * radius
+        ):
+            zoom += 1
     return zoom - 1  # Go one zoom level back
+
+
+def get_bounding_box(locations):
+    """
+    Calculate the minimum and maximum latitude and longitude
+    from the given set of coordinates to form a bounding box
+
+    :Parameters:
+        `locations`: List of tuples containing latitude and longitude.
+    """
+    min_lat = min(locations, key=lambda x: x[0])[0]
+    max_lat = max(locations, key=lambda x: x[0])[0]
+    min_lon = min(locations, key=lambda x: x[1])[1]
+    max_lon = max(locations, key=lambda x: x[1])[1]
+    return min_lat, max_lat, min_lon, max_lon
+
+
+def get_bounding_box_center(locations):
+    """
+    Find the center of this bounding box by averaging the
+    minimum and maximum latitudes and longitudes
+
+    :Parameters:
+        `locations`: List of tuples containing latitude and longitude.
+    """
+    min_lat, max_lat, min_lon, max_lon = get_bounding_box(locations)
+    center_lat = (min_lat + max_lat) / 2
+    center_lon = (min_lon + max_lon) / 2
+    return center_lat, center_lon
+
+
+def get_fit_zoom_level(locations, map_width, map_height):
+    """
+    Calculates the zoom level to fit all locations into the map view.
+
+    Determine the zoom level that fits the bounding box within the map view.
+    This involves calculating the required scale to fit both the width
+    and height of the bounding box into the viewport.
+
+    :Parameters:
+        `locations`: List of tuples containing latitude and longitude.
+        `map_width`: Width of the map
+        `map_height`: Height of the map
+
+    :return: Calculated zoom level.
+    """
+    min_lat, max_lat, min_lon, max_lon = get_bounding_box(locations)
+
+    # map_width /= 5
+    # map_height /= 5
+
+    # Constants for zoom calculation
+    TILE_SIZE = 256
```

Comment:
Please pass it as a parameter, with a default of 256 (use `tile_size=256`), as the library can be used with other source size (512 for retina screen for eg)